java/src/main/java/com/azure/graph/bulk/impl/model/GremlinEdgeVertexInfo.java

```diff
@@ -90,7 +99,17 @@ public boolean equals(Object o) {
 
         GremlinEdgeVertexInfo c = (GremlinEdgeVertexInfo) o;
 
-        return Objects.equals(c.getId(), this.getId());
+        if (isNotEqual(c.id, this.id)) return false;
```

Comment:
Update hashCode as well ? Should be about to use Objects.hash(fieldA, fieldB, ...)